agent/bench-scripts/pbench-uperf

```diff
@@ -181,280 +182,295 @@ function gen_xml {
+			fi
 		;;
 		-t|--test-types)
-		shift
-		if [ -n "$1" ]; then
-			test_types="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				test_types="$1"
+				shift
+			fi
 		;;
 		--tool-group)
-		shift
-		if [ -n "$1" ]; then
-			tool_group="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				tool_group="$1"
+				shift
+			fi
 		;;
 		-m|--message-sizes)
-		shift
-		if [ -n "$1" ]; then
-			message_sizes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				message_sizes="$1"
+				shift
+			fi
 		;;
 		-r|--runtime)
-		shift
-		if [ -n "$1" ]; then
-			runtime="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				runtime="$1"
+				shift
+			fi
 		;;
 		--log-response-times)
-		shift
-		if [ -n "$1" ]; then
-			log_response_times="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				log_response_times="$1"
+				shift
+			fi
 		;;
 		-p|--protocols)
-		shift
-		if [ -n "$1" ]; then
-			protocols="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				protocols="$1"
+				shift
+			fi
 		;;
 		-C|--client|--clients)
-		shift
-		if [ -n "$1" ]; then
-			clients="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				clients="$1"
+				shift
+			fi
 		;;
 		-S|--server|--servers)
-		shift
-		if [ -n "$1" ]; then
-			servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				servers="$1"
+				shift
+			fi
 		;;
 		--vsock-server|--vsock-servers)
-		shift
-		if [ -n "$1" ]; then
-			vsock_servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				vsock_servers="$1"
+				shift
+			fi
 		;;
 		-c|--config)
-		shift
-		if [ -n "$1" ]; then
-			config="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				config="$1"
+				shift
+			fi
 		;;
 		--start-iteration-num)
-		shift
-		if [ -n "$1" ]; then
-			start_iteration_num=$1
-			shift
-		fi
+			if [ -n "$1" ]; then
+				start_iteration_num=$1
+				shift
+			fi
 		;;
 		--client-node|--client-nodes)
-		shift
-		if [ -n "$1" ]; then
-			client_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				client_nodes="$1"
+				shift
+			fi
 		;;
 		--server-node|--server-nodes)
-		shift
-		if [ -n "$1" ]; then
-			server_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				server_nodes="$1"
+				shift
+			fi
 		;;
 		--sysinfo)
-		shift;
-		if [ -n "$1" ]; then
-			sysinfo="$1"
-			shift;
-		fi
+			if [ -n "$1" ]; then
+				sysinfo="$1"
+				shift
+			fi
 		;;
 		--)
-		shift
-		break
+			break
 		;;
 		*)
-		error_log "[$script_name] bad option, \"$1 $2\""
-		exit 1
-		break
+			error_log "[$script_name] bad option, \"${arg} ${1}\""
+			usage >&2
+			exit 1
 		;;
 	esac
 done
 verify_common_bench_script_options $tool_group $sysinfo
 
-# Verify the number of clients and servers match.
-if [ ! -z "$clients" -a "$postprocess_only" != "y" ]; then
+clients=${clients:-${clients_def}}
+servers=${servers:-${servers_def}}
+
+if [[ "${postprocess_only}" != "y" ]]; then
+	# Boolean value recording if local clients or servers are specified.
+	let check_local=0
+
+	# Verify that the number of clients, servers, and optional vsock servers
+	#  match.
```

Comment:
Did you mean to indent `match` two spaces instead of one? It looks inconsistent. 😉